tools/bin/ariaExtract.py

```diff
@@ -142,6 +142,34 @@ def main():
     logging.basicConfig(level=log_level, format=ARIAtools.util.log.FORMAT)
     LOGGER.info('Extract Product Function')
 
+    # Check whether all necessary inputs were specified.
+    # some products require a DEM to extract -- if any of those are requested,
+    # ensure that a valid DEM is specified
+    if args.layers:
+        # format list of layers
+        layers = list(args.layers.split(','))
+        layers = [i.replace(' ', '') for i in layers]
+        layers = ['all' if layer.lower() == 'all'
+                  else layer for layer in layers]
+
+        # list of layers requiring DEM for extraction
+        layers_requiring_dem = ['all',
+                                'bPerpendicular',
+                                'bParallel',
+                                'incidenceAngle',
+                                'lookAngle',
+                                'azimuthAngle',
+                                'solidEarthTide']
+
+        # check that DEM is specified depending on layers requested
+        intersecting_layers = [layer for layer in layers
```

Comment:
IMO let's use sets and intersection here (more expressive):
`if len(set(LAYERS_REQUIRING_DEM).intersection(layers)) == 0:`
or declare LAYERS_REQUIRING_DEM as a set (use curly braces), then it will be 
`if len(LAYERS_REQUIRING_DEM.intersection(layers)) == 0:`